### ISSUE TITLE
Support iproute2 bpf_map_def layout

### ISSUE THREAD

**Issue by alessandrod** [alessandrod/aya#10]
This is how aya currently defines `bpf_map_def`:  https://github.com/alessandrod/aya/blob/bb595c4e69ff0c72c8327e7f64d43ca7a4bc16a3/aya/src/bpf.rs#L52-L60

Many older eBPF programs use an old definition of `bpf_map_def` that didn't include the `id`  and `pinning` fields. Currently aya will fail to parse those programs because of this check: https://github.com/alessandrod/aya/blob/bb595c4e69ff0c72c8327e7f64d43ca7a4bc16a3/aya/src/obj/mod.rs#L512-L517

We could easily support those older programs by relaxing the above check a little - checking that the struct contains up to the `map_flags` field, and set `id`  and `pinning` to `0`.

**Comment by RafaelOrtizRC** [comment]
Where are the `id` and `pinning` fields coming from? Here's the anonymous struct in the latest 5.12:

```c
	struct { /* anonymous struct used by BPF_MAP_CREATE command */
		__u32	map_type;	/* one of enum bpf_map_type */
		__u32	key_size;	/* size of key in bytes */
		__u32	value_size;	/* size of value in bytes */
		__u32	max_entries;	/* max number of entries in a map */
		__u32	map_flags;	/* BPF_MAP_CREATE related
					 * flags defined above.
					 */
		__u32	inner_map_fd;	/* fd pointing to the inner map */
		__u32	numa_node;	/* numa node (effective only if
					 * BPF_F_NUMA_NODE is set).
					 */
		char	map_name[BPF_OBJ_NAME_LEN];
		__u32	map_ifindex;	/* ifindex of netdev to create on */
		__u32	btf_fd;		/* fd pointing to a BTF type data */
		__u32	btf_key_type_id;	/* BTF type_id of the key */
		__u32	btf_value_type_id;	/* BTF type_id of the value */
		__u32	btf_vmlinux_value_type_id;/* BTF type_id of a kernel-
						   * struct stored as the
						   * map value
						   */
	};
```

**Comment by alessandrod** [comment]
> Where are the `id` and `pinning` fields coming from? Here's the anonymous struct in the latest 5.12:
> 

iproute2, see https://github.com/alessandrod/aya/pull/14#issuecomment-865325330

**Comment by RafaelOrtizRC** [comment]
Closed by #14
